kubeshop/testkube-dashboard
# fix: clean up API endpoint code and resolve race conditions

## Changes

- `apiEndpoint` is taken only from `services/apiEndpoint.ts` - cached or `localStorage`
   - before, depending on the function, it's been taken from `MainContext`, Redux store, or directly from `localStorage`
- Race conditions have been solved
   - Before, when `apiEndpoint` query parameter was used, there was a race between setting `apiEndpoint` from query param, and saving the verified one from the `localStorage`. The query param from `apiEndpoint` was replaced, as it was saved earlier.
- Clean up the code related to the API endpoint
   - Unify `EndpointModal.tsx` and `ApiEndpoint.tsx` logic
   - Move `<EndpointModal>` rendering from `<Sider>` to `<App>`
   - Move logic for sanity API endpoint check from `EndpointModal` to the `App` component
      - Keeping that logic in the `EndpointModal` was obscure, as it was global logic
      - It helped to avoid saving in the `localStorage` when the user took it from the environment variables

## Fixes

- https://github.com/kubeshop/testkube/issues/2779

## How to test it

- There are some unit tests added
- Check if everything works well
- Check if `apiEndpoint` query param replaces value from the `localStorage`

## Checklist
